state-synchronizer/src/synchronizer.rs

```diff
@@ -132,12 +151,17 @@ impl StateSyncClient {
         &self,
         // *successfully* committed transactions
         committed_txns: Vec<Transaction>,
+        reconfiguration_events: Vec<ContractEvent>,
```

Comment:
this probably would return all the events and let state sync to do the filtering? also why does it need full list of transactions? isn't id enough?

Reply:
The way this function is called, consensus passes in the already-filtered events. I'm using `Vec` to avoid `None`/`Some` semantics.. if that is preferred I can switch to None/Some

Reply:
If we're going to have multiple different events, the subscription service probably should be responsible to filter out events since only it knows about what events are subscribed.